include/range/v3/view/unique.hpp

```diff
@@ -33,27 +33,44 @@ namespace ranges
         {
             struct unique_fn
             {
-                template<typename Rng>
+            private:
+                friend view_access;
+                template<typename C,
+                        CONCEPT_REQUIRES_(!Range<C>())>
+                static auto bind(unique_fn unique, C pred)
+                RANGES_DECLTYPE_AUTO_RETURN
+                (
+                    std::bind(unique, std::placeholders::_1, protect(std::move(pred)))
+                )
+
+            public:
+                template<typename Rng, typename C = equal_to,
+                        typename I = iterator_t<Rng>>
                 using Concept = meta::and_<
                     ForwardRange<Rng>,
-                    EqualityComparable<range_value_type_t<Rng>>>;
+                    IndirectRelation<C, I>>;
```

Comment:
I suggest using `iterator_t<Rng>` directly here:

```suggestion
                    IndirectRelation<C, iterator_t<Rng>>>;
```
 and avoiding the extra template parameter to save a few microseconds of compile time.

Reply:
Should this be:
```suggestion
                    IndirectRelation<C&, I>>;
```
I literally don't recall if `IndirectRelation` is value category sensitive.

Reply:
Every where that I saw on the code it appear as just `C`
See `algorithm/find.hpp:45`